Python testing: Fix reporting on setup_class error

cecille opened the pull request:
Also add error text to make the error easier to find.

Fixes: https://github.com/project-chip/connectedhomeip/issues/34959

cecille committed:
Python testing: Fix reporting on setup_class error

Also add error text to make the error easier to find.

restyled-commits committed:
Restyled by isort

cecille committed:
Fix lint

woody-apple reviewed (approved):
Pre-approving so this doesn't get blocked.

cecille committed:
Change exemption to not carry chip_error reference

chip_error is a ctypes struct with a const char* pointer internally.
This cannot be pickled, so it's causing problems with the mobly
framework.

cecille committed:
Fix some tests using removed class member
